graveler/committed/manager.go

```diff
@@ -106,6 +107,7 @@ func (c *committedManager) Apply(ctx context.Context, ns graveler.StorageNamespa
 	if err != nil {
 		return "", fmt.Errorf("get metarange ns=%s id=%s: %w", ns, rangeID, err)
 	}
+	defer metaRangeIterator.Close()
```

Comment:
🙇

Reply:
Resource management is too hard for human beings.  If only we had something like finalizers, but with deterministic  execution, automatic resource management would be so easy...  We could call them something like "destroyers" or "anticonstructors".

Reply:
maybe we can take the http request Close linter and make it work on our iterators

Reply:
I shall have a look!